internal/controller/awsvalidator_controller.go

```diff
@@ -68,8 +62,8 @@ func (r *AwsValidatorReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	// Ensure both AWS env vars are set.
-	if err := r.configureAwsAuth(validator.Spec.Auth, req.Namespace, l); err != nil {
+	// Override auth data in spec with auth data from Secret if applicable.
+	if validator.Spec.Auth, err = r.authFromSecret(validator.Spec.Auth, req.Namespace, l); err != nil {
```

Comment:
You should validate the auth here in Reconcile after the override. Otherwise the behaviour is inconsistent between k8s and direct modes.